src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,138 @@
+package duke;
+
+import duke.Deadline;
+import duke.Event;
+import duke.Task;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+
+public class TaskList {
+
+    public ArrayList<Task> myList;
+
+    /**
+     * initialises the TaskList (supposed to have loading functionality, but will add on later)
+     */
+
+    public TaskList(ArrayList<Task> parsedList) {
+        myList = parsedList;
+    }
+
+    /**
+     * calls an appropriate method based on the input param
+     * @param input
+     */
+
+    public void action(String input) {
+        if (input.equals("list")) {
+            listTasks();
+        } else if (input.startsWith("done")) {
+            markAsDone(input);
+        } else if (input.startsWith("delete")) {
+            deleteTask(input);
+        } else if (input.startsWith("find")) {
+            findTask(input);
+        } else {
+            addTask(input);
+        }
+    }
+
+    /**
+     * forms a String of all the tasks, for saving into file
+     * @return String
+     */
+    
+    public String output() {
+        String output = "";
+        for (int i = 0; i < myList.size(); i++) {
+            output = output + myList.get(i) + "\n";
+        }
+        return output;
+    }
+
+    private void listTasks() {
+        System.out.println("Here is your to-do list!");
+        for (int i = 0; i < myList.size(); i++) {
+            System.out.println(i + 1 + ". " + myList.get(i).toString());
+        }
+    }
+
+    private void addTask(String input) {
+        try {
+            if (input.startsWith("todo")) {
```

Comment:
Perhaps you could use an enum to represent the different tasks and a switch statement here?